confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,701 @@
+            level (str): Compatibility level. See API reference for a list of
+                valid values.
+            subject_name (str, optional): Subject to update. Sets compatibility
+                level policy if not set.
+
+        Returns:
+            str: The newly configured compatibility level.
+
+        Raises:
+            SchemaRegistryError: If the compatibility level is invalid.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#put--config
+
+        """  # noqa: E501
+        if level is None:
+            raise ValueError("level must be set")
+
+        if subject_name is None:
+            return self._rest_client.put('config',
+                                         body={'compatibility': level.upper()})
+
+        return self._rest_client.put('config/{}'
+                                     .format(_urlencode(subject_name)),
+                                     body={'compatibility': level.upper()})
+
+    def get_compatibility(self, subject_name=None):
+        """
+        Get the current compatibility level.
+
+        Args:
+            subject_name (str, optional): Subject name. Returns global policy
+                if left unset.
+
+        Returns:
+            str: Compatibility level for the subject if set, otherwise the
+                global compatibility level.
+
+        Raises:
+            SchemaRegistryError: if the request was unsuccessful.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--config
+
+        """  # noqa: E501
+        if subject_name is not None:
+            url = 'config/{}'.format(_urlencode(subject_name))
+        else:
+            url = 'config'
+
+        return self._rest_client.get(url)
+
+
+class Schema(object):
+    """
+    An unregistered Schema.
+
+    Args:
+        schema_str (str): String representation of the schema.
+        references ([Schema]): Schemas referenced by this schema.
+        schema_type (str): The schema type: AVRO, PROTOBUF or JSON.
+
+    """
+    __slots__ = ['schema_str', 'references', 'schema_type']
+
+    def __init__(self, schema_str, schema_type, references=[]):
+        super(Schema, self).__init__()
+
+        self.schema_str = schema_str
+        self.schema_type = schema_type
+        self.references = references
+
+    def __eq__(self, other):
+        return str(self) == str(other)
+
+
+class RegisteredSchema(object):
+    """
+    Schema registration information.
+
+    As of Confluent Platform 5.5 Schema's may now hold references to other
+    registered schemas.
+
+    The class represents a reference to another schema which is used by this
+    schema. As long as there is a references to a schema it may not be deleted.
+
+    Args:
+        schema_id (int): Registered Schema id
+        schema_type (str): Type of schema
+        schema (Schema): Registered Schema
```

Comment:
`schema_type` is duplicated - it's at the top level and in `Schema`. I would have `schema_str` here instead of `schema`, include `references` at the top level and (optionally) add a `@property` exposing `Schema` derived from the other values.